### ISSUE TITLE
write_state cannot be used in pure_localization mode?

### ISSUE THREAD

**Issue by duchengyao** [googlecartographer/cartographer_ros#494]
when `TRAJECTORY_BUILDER.pure_localization` set `true` and I'm using `rosservice call /write_state aaa.bag.pbstream`, I found the map file become smaller (about 10% than before, and I can use this small map next time)

But when I'm continue driving for a bit, cartographer match the full map and robot find the place in map, I'm calling  `rosservice call /write_state aaa.bag.pbstream` again. ROS crash

```
u@u:~$ rosservice call /write_state ~/catkin/src/S1NH/configuration/map.bag.pbstream
ERROR: transport error completing service call: unable to receive data from sender, check sender's logs for details
```

```
[cartographer_node-12] process has died [pid 31108, exit code -11, cmd /home/u/catkin/install_isolated/lib/cartographer_ros/cartographer_node -configuration_directory /home/u/catkin/src/S1NH/configuration -configuration_basename cartographer_turtlebot_lidar_2d.lua -map_filename /home/u/catkin/src/S1NH/configuration/map.bag.pbstream scan:=/scan __name:=cartographer_node __log:=/home/u/.ros/log/943e600e-8e33-11e7-b80d-00c2c6f590d3/cartographer_node-12.log].
log file: /home/u/.ros/log/943e600e-8e33-11e7-b80d-00c2c6f590d3/cartographer_node-12*.log
[ WARN] [1504173923.207725848]: Costmap2DROS transform timeout. Current time: 1504173923.2076, global_pose stamp: 1504173921.1870, tolerance: 2.0000
[ WARN] [1504173923.207808130]: Could not get robot pose, cancelling pose reconfiguration
```

**Comment by SirVer** [comment]
Your use case sounds strange to me too - why would you want to save the state of pure localization? This will be essentially the same as the map you originally loaded into Cartographer. Could you elaborate a bit what you are trying to achieve?

This should obviously not crash, so this sounds like a bug, but I did not yet try to reproduce it.

**Comment by wohe** [comment]
Should be fixed by #550.
